pkg/resources/meshexpansion/multimesh.go

```diff
@@ -107,19 +117,19 @@ func (r *Reconciler) multimeshEnvoyFilter(selector map[string]string) *k8sutil.D
 	}
 }
 
-func (r *Reconciler) multimeshDestinationRule() *k8sutil.DynamicObject {
+func (r *Reconciler) multimeshDestinationRule(domain string) *k8sutil.DynamicObject {
 	return &k8sutil.DynamicObject{
 		Gvr: schema.GroupVersionResource{
 			Group:    "networking.istio.io",
 			Version:  "v1alpha3",
 			Resource: "destinationrules",
 		},
 		Kind:      "DestinationRule",
-		Name:      r.Config.WithRevision(multimeshResourceNamePrefix + "-destinationrule"),
+		Name:      r.Config.WithRevision(multimeshResourceNamePrefix + "-" + strings.ReplaceAll(domain, ".", "-")),
```

Comment:
My earlier suggestion to allow `-` in a domain name clashes with this. Multiple domains could be mapped to the same destinationrule name. So, `-` should either be kept as disallowed, or maybe use `GenerateName` here?

Reply:
I changed it to use crc32 hash as suffix.